### DISCUSSION TITLE
Sequel 5.81.0 Released

### DISCUSSION THREAD

**Discussion by jeremyevans** [jeremyevans/sequel#2172] [General]
Sequel 5.81.0 has been released!

# New Features

* A temporarily_release_connection Database extension has been added,
  designed for multithreaded transactional testing.

  This allows one thread to start a transaction, and then release
  the connection back for usage by the connection pool, so that
  other threads can operate on the connection object safely inside
  the transaction.  This requires the connection pool be limited
  to a single connection, to ensure that the released connection
  can be reacquired.  It's not perfect, because if the connection
  is disconnected and removed from the pool while temporarily
  released, there is no way to handle that situation correctly.
  Example:

  ```ruby
    DB.transaction(rollback: :always, auto_savepoint: true) do |conn|
      DB.temporarily_release_connection(conn) do
        # Other threads can operate on connection safely inside
        # the transaction
        yield
      end
    end
   ```

# Other Improvements

* In the caller_logging and provenance extensions, Ruby internal
  caller locations are skipped when trying to locate the appropriate
  caller line to include.

* A couple ignored block warnings in plugin apply methods have been
  fixed on Ruby 3.4.

Thanks,
Jeremy

**Comment by janko** [comment]
I see the new extension directly solves the use case we discussed in https://github.com/jeremyevans/sequel/discussions/1959, thank you!

**Reply by jeremyevans** [reply]
Yep, it took me quite a long time to think of a design I was comfortable with to allow this.  Thanks for putting the idea in my mind.
